### ISSUE TITLE
SS4 FulltextSearchable returns errors

### ISSUE THREAD

**Issue by fonsekaean** [tractorcow/silverstripe-fluent#360]
After installing Fluent over a site which has FulltextSearchable::enabled the search forms starts throwing errors. 

```
[Emergency] Uncaught SilverStripe\ORM\Connect\DatabaseException: Couldn't run query: SELECT DISTINCT count(*) FROM "SiteTree_Live" WHERE ( MATCH (Title, MenuTitle, Content, MetaDescription) AGAINST ('hello*' IN BOOLEAN MODE) + MATCH (Title, MenuTitle, Content, MetaDescription) AGAINST ('hello*' IN BOOLEAN MODE) AND ShowInSearch 0) AND (("SiteTree_Localised_en_US"."ID" IS NOT NULL)) 42S22-1054: Unknown column 'SiteTree_Localised_en_US.ID' in 'where clause'
```

The above is the example error. 

I debugged it to find out the cause of the issue. 

* In FluentExtension.php around the lines 329 - 340 it joins the SiteTree table with SiteTree_Localised which also has an alias SiteTree_Localised_en_US
* When performing a search MySQLDatabase.php line line 265 it replaces the from with this function call `$query->setFrom($sqlTables[$class]);` that makes the query to lose the alias and throws an error. 

I'd be happy to send a patch for this, but before that wants to check on whats the best way to sort this. Either to make the MySQLDatabase.php extendable and have its own extension hooks to edit this query. Or should the MySQLDatabase.php be overridden completely within the Fluent module ? 

Thanks

**Comment by robbieaverill** [comment]
Hey @fonsekaean, thanks for reporting.

I'll defer to @tractorcow on the suggested fix for this.

FYI we're just adding support for the [fulltextsearch](https://github.com/silverstripe/silverstripe-fulltextsearch) module in #359

**Comment by fonsekaean** [comment]
Thanks

**Comment by tractorcow** [comment]
Yeah fixing fulltextsearch for this module is definitely on the "todo" list.

We will likely never "natively" support the default fulltextsearch behaviour in mysqldatabase. We may proxy that method and inject a custom search code.

@robbieaverill another candidate for https://github.com/tractorcow/silverstripe-proxy-db :D

**Comment by chromos33** [comment]
Any progress on this?

**Comment by DorsetDigital** [comment]
Sorry to be a pain, but I'm a bit confused by the status of this one.  A similar issue was reported on #335 (which has been closed).  Do the changes in #359 resolve this issue?  If so, is there an indication on when it will be released?   Just trying to work out my search strategy for a couple of sites that need to go live imminently.

**Comment by robbieaverill** [comment]
Hey @DorsetDigital, in lieu of being able to give you a timeline estimate for this issue, #359 added support for Solr indexing. This issue is is tracking compatibility with the default search engine in SilverStripe core.

**Comment by innomedia** [comment]
Any news on this?

**Comment by xini** [comment]
Any updates on this?

**Comment by baukezwaan** [comment]
Does anyone know the status of this? Or a work-around maybe?

**Comment by robbieaverill** [comment]
@sunnysideup has a workaround for this issue. Would you kind sharing it here?

**Comment by sunnysideup** [comment]
I am sorry, but I am at a complete loss on how we fixed this. 

FULLTEXTSEARCH + SS4 + MYSQL? 

This worked for us - AFAIK. Sorry, I realise I may be barking up the wrong forest ;-)

![image](https://user-images.githubusercontent.com/167154/56193122-05c11400-6084-11e9-97c7-40215acdad89.png)

PS this is an awesome module for search: https://github.com/sunnysideup/silverstripe-search_simple_smart.... well, from my perspective.

**Comment by baukezwaan** [comment]
The $indexes static doesn't seem to be the problem here (or at least it didn't fixed it). 
The augmentSQL-function from FluentExtension.php appends the locale to the table name (resulting in non-existing table like `SiteTree_Localised_en_US`).

Maybe @fonsekaean, have you got it running eventually?


(PS @sunnysideup that module looks really awesome! Will give it a swing soon...)

**Comment by taoceanz** [comment]
@baukezwaan Thanks for the work around.

**Comment by taoceanz** [comment]
For anyone who's needing core search functionality with Fluent, we're currently working on a PR to integrate a SS Core search extension into the Fluent module. Base extension taken from @baukezwaan gist.
https://github.com/tractorcow-farm/silverstripe-fluent/pull/547

**Comment by Misagh90** [comment]
Is there any update on this issue?

**Comment by robbieaverill** [comment]
#547 adds some support for searching across locales, but doesn't allow you to restrict search results to a particular locale.

**Comment by taoceanz** [comment]
@robbieaverill could you make a new tag on Fluent for PR #547 to be able to be required via Composer?
By restricting search to a particular locale, do you mean preventing searching when in a particular locale? From my testing the Fluent Search form _should_ return results for only the current locale.

**Comment by robbieaverill** [comment]
@taoceanz perhaps I tested it wrong, but I localised my home page in two locales and was getting a search hit in both locales while searching for the other locale's home page content. I assume it would work correctly when a page doesn't exist in both locales.

Re: a using the new changes in composer, you can use the `4.5.x-dev` constraint to pull the latest changes in, e.g. `"require": { "tractorcow/silverstripe-fluent": "4.5.x-dev" }`

**Comment by taoceanz** [comment]
@robbieaverill fair play. I believe search results return only pages available in the current locale, but maybe I'm wrong.

Thanks for noting the dev link. I've got minimum stability set to 'stable' to avoid dev versions. Do you know when there'd be enough changes ready to tag v4.4.3?
